2009-02-20  David S. Miller  <davem@davemloft.net>
 
	* sysdeps/sparc/sparc64/memchr.S: Use unsigned branch on greater to
	test end of buffer.

---
 sysdeps/sparc/sparc64/memchr.S |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/sysdeps/sparc/sparc64/memchr.S
+++ b/sysdeps/sparc/sparc64/memchr.S
@@ -96,7 +96,7 @@
 							 * %o4 =   value XOR c		*/
 2:	cmp		%o0, %o2			/* IEU1		Group		*/
 
-	bg,pn		%XCC, 11f			/* CTI				*/
+	bgu,pn		%XCC, 11f			/* CTI				*/
 	 ldxa		[%o0] ASI_PNF, %o3		/* Load				*/
 	sub		%o4, %g1, %o5			/* IEU0		Group		*/
 	add		%o0, 8, %o0			/* IEU1				*/
